lib/ansible/modules/cloud/docker/docker_node.py

```diff
@@ -0,0 +1,301 @@
+
+- name: Remove all labels assigned to node
+  docker_node:
+    hostname: mynode
+    labels:
+    labels_state: replace
+
+- name: Remove node labels
+  docker_node:
+    hostname: mynode
+    labels_state: remove
+'''
+
+RETURN = '''
+node_facts:
+  description: Information about node after 'update' operation
+  returned: success
+  type: dict
+
+'''
+
+try:
+    from docker.errors import APIError
+except ImportError:
+    # missing docker-py handled in ansible.module_utils.docker_common
+    pass
+
+from ansible.module_utils.docker_common import (
+    DockerBaseClass,
+)
+
+from ansible.module_utils._text import to_native
+
+from ansible.module_utils.docker_swarm import AnsibleDockerSwarmClient
+
+
+class TaskParameters(DockerBaseClass):
+    def __init__(self, client):
+        super(TaskParameters, self).__init__()
+
+        # Spec
+        self.name = None
+        self.labels = None
+        self.labels_state = None
+
+        # Node
+        self.availability = None
+        self.role = None
+
+        for key, value in client.module.params.items():
+            setattr(self, key, value)
+
+
+class SwarmNodeManager(DockerBaseClass):
+
+    def __init__(self, client, results):
+
+        super(SwarmNodeManager, self).__init__()
+
+        self.client = client
+        self.results = results
+        self.check_mode = self.client.check_mode
+
+        self.client.fail_task_if_not_swarm_manager()
+
+        self.parameters = TaskParameters(client)
+
+        self.node_update()
+
+    def node_update(self):
+        if not (self.client.check_if_swarm_node(node_id=self.parameters.hostname)):
+            self.client.fail(msg="This node is not part of a swarm.")
+            return
+
+        try:
+            status_down = self.client.check_if_swarm_node_is_down()
+        except APIError:
+            return
+
+        if status_down:
+            self.client.fail(msg="Can not update the node. The status node is down.")
+
+        try:
+            node_info = self.client.inspect_node(node_id=self.parameters.hostname)
+        except APIError as exc:
+            self.client.fail(msg="Failed to get node information for %s" % to_native(exc))
+
+        __changed = False
+        node_spec = dict(
+            Availability=self.parameters.availability,
+            Role=self.parameters.role,
+            Labels=self.parameters.labels,
+        )
+
+        if self.parameters.role is None:
+            node_spec['Role'] = node_info['Spec']['Role']
+
+        if self.parameters.availability is None:
+            node_spec['Availability'] = node_info['Spec']['Availability']
+
+        if self.parameters.labels_state == 'replace':
+            node_spec['Labels'] = self.parameters.labels
+            __changed = True
+
+        if self.parameters.labels_state == 'remove':
+            node_spec['Labels'] = node_info['Spec']['Labels']
+
+            if self.parameters.labels is None and node_info['Spec']['Labels'] is not None:
```

Comment:
```suggestion
            if self.parameters.labels is None and not node_info['Spec']['Labels']:
```
It could also be empty.